tests/evm/test_oog_call.py

```diff
@@ -1,127 +1,172 @@
-            callees, call_contexts, stacks, is_warm_accesss
+        for opcode, caller_context, stack, is_warm_access in product(
+            call_opcodes, call_contexts, stacks, is_warm_accesses
         )
     ]
 
 
 TESTING_DATA = gen_testing_data()
 
 
-@pytest.mark.parametrize("caller, callee, caller_ctx, stack, is_warm_access", TESTING_DATA)
-def test_root_call(
-    caller: Account,
+@pytest.mark.parametrize(
+    "callee, caller_bytecode, caller_context, stack, has_value, is_warm_access", TESTING_DATA
+)
+def test_oog_call_root(
     callee: Account,
-    caller_ctx: CallContext,
+    caller_bytecode: Bytecode,
+    caller_context: CallContext,
     stack: Stack,
+    has_value: bool,
     is_warm_access: bool,
 ):
     randomness = rand_fq()
-    caller_bytecode = (
-        Bytecode()
-        .call(
-            stack.gas,
-            callee.address,
-            stack.value,
-            stack.cd_offset,
-            stack.cd_length,
-            stack.rd_offset,
-            stack.rd_length,
-        )
-        .stop()
-    )
     caller_bytecode_hash = RLC(caller_bytecode.hash(), randomness)
     callee_bytecode_hash = RLC(callee.code_hash(), randomness)
 
     is_success = False
+    program_counter = 231 if has_value else 198
 
-    # fmt: off
     rw_dictionary = (
         RWDictionary(24)
         .call_context_read(1, CallContextFieldTag.TxId, 1)
-        .stack_read(1, 1017, RLC(stack.gas, randomness))
-        .stack_read(1, 1018, RLC(callee.address, randomness))
-        .stack_read(1, 1019, RLC(stack.value, randomness))
-        .stack_read(1, 1020, RLC(stack.cd_offset, randomness))
-        .stack_read(1, 1021, RLC(stack.cd_length, randomness))
-        .stack_read(1, 1022, RLC(stack.rd_offset, randomness))
-        .stack_read(1, 1023, RLC(stack.rd_length, randomness))
-        .stack_write(1, 1023, RLC(is_success, randomness))
-        .account_read(callee.address, AccountFieldTag.CodeHash, callee_bytecode_hash)
-        .tx_access_list_account_read(1, callee.address, is_warm_access)
-        .call_context_read(1, CallContextFieldTag.IsSuccess, 0)
+        .stack_read(1, 1018 - has_value, RLC(stack.gas, randomness))
+        .stack_read(1, 1019 - has_value, RLC(callee.address, randomness))
     )
-    # fmt: on
+    if has_value:
+        rw_dictionary.stack_read(1, 1019, RLC(stack.value, randomness))
+    # fmt: off
+    rw_dictionary \
+        .stack_read(1, 1020, RLC(stack.cd_offset, randomness)) \
+        .stack_read(1, 1021, RLC(stack.cd_length, randomness)) \
+        .stack_read(1, 1022, RLC(stack.rd_offset, randomness)) \
+        .stack_read(1, 1023, RLC(stack.rd_length, randomness)) \
+        .stack_write(1, 1023, RLC(is_success, randomness)) \
+        .account_read(callee.address, AccountFieldTag.CodeHash, callee_bytecode_hash) \
+        .tx_access_list_account_read(1, callee.address, is_warm_access) \
+        .call_context_read(1, CallContextFieldTag.IsSuccess, 0)
+    # fmt on
 
```

Comment:
fmt on or fmt: on?

Reply:
Fixed to `fmt: on` in both line 169 and 260 in [94fe42d](https://github.com/privacy-scaling-explorations/zkevm-specs/pull/368/commits/94fe42d09f76b84ef3fe90cf556b9926046638ae)